components/integrations/installed_incoming_webhook.tsx

```diff
@@ -31,45 +43,47 @@ export function matchesFilter(incomingWebhook, channel, filter) {
     return false;
 }
 
-export default class InstalledIncomingWebhook extends React.PureComponent {
-    static propTypes = {
-
-        /**
-        * Data used for showing webhook details
-        */
-        incomingWebhook: PropTypes.object.isRequired,
-
-        /**
-        * Function to call when webhook delete button is pressed
-        */
-        onDelete: PropTypes.func.isRequired,
-
-        /**
-        * String used for filtering webhook item
-        */
-        filter: PropTypes.string,
-
-        /**
-        * Data used for showing created by details
-        */
-        creator: PropTypes.object.isRequired,
-
-        /**
-        *  Set to show available actions on webhook
-        */
-        canChange: PropTypes.bool.isRequired,
-
-        /**
-        *  Data used in routing of webhook for modifications
-        */
-        team: PropTypes.object.isRequired,
-
-        /**
-        *  Data used for filtering of webhook based on filter prop
-        */
-        channel: PropTypes.object,
-    }
+type Props = {
+
+    /**
+     * Data used for showing webhook details
+     */
+    incomingWebhook: IncomingWebhook;
+
+    /**
+     * Function to call when webhook delete button is pressed
+     */
+    onDelete: (incomingWebhook: IncomingWebhook) => void;
+
+    /**
+     * String used for filtering webhook item
+     */
+    filter?: string;
+
+    /**
+     * Data used for showing created by details
+     */
+    creator?: {
```

Comment:
(question) why is creator optional? in the original propTypes it is a required.

Reply:
Mostly laziness in trying to get the tests to compile correctly. I'll spend some time today and work through updating those test flies.